src/java.base/share/native/libjli/args.c

```diff
@@ -354,42 +354,38 @@ static JLI_List readArgFile(FILE *file) {
     return rv;
 }
 
+static void reportAndExit(const char* fmt, const char* arg) {
+    if (fmt != NULL) JLI_ReportMessage(fmt, arg);
+    exit(1);
+}
+
 /*
  * if the arg represent a file, that is, prefix with a single '@',
  * return a list of arguments from the file.
  * otherwise, return NULL.
  */
 static JLI_List expandArgFile(const char *arg) {
-    FILE *fptr;
-    struct stat st;
     JLI_List rv;
+    struct stat st;
+    FILE *fptr = fopen(arg, "r");
 
-    /* failed to access the file */
-    if (stat(arg, &st) != 0) {
-        JLI_ReportMessage(CFG_ERROR6, arg);
-        exit(1);
-    }
-
-    if (st.st_size > MAX_ARGF_SIZE) {
-        JLI_ReportMessage(CFG_ERROR10, MAX_ARGF_SIZE);
-        exit(1);
-    }
-
-    fptr = fopen(arg, "r");
     /* arg file cannot be openned */
-    if (fptr == NULL) {
-        JLI_ReportMessage(CFG_ERROR6, arg);
-        exit(1);
+    if (fptr == NULL || fstat(fileno(fptr), &st) != 0) {
+        reportAndExit(CFG_ERROR6, arg);
+    } else {
+        if (st.st_size > MAX_ARGF_SIZE) {
+            JLI_ReportMessage(CFG_ERROR10, MAX_ARGF_SIZE);
+            reportAndExit(NULL, NULL);
```

Comment:
This should be just one statement,
    reportAndExit(CFG_ERROR10, MAX_ARGF_SIZE);
or?

Reply:
I think that did not work because it does not fit the param types of reportAndExit but I can simplify it otherwise.

Reply:
Ah, I see.
Maybe it's a bit bike-sheddy but as you're using reportAndExit only twice now, wouldn't it be less convoluted if you use
```
JLI_ReportMessage(...);
exit(1);
```
inline in both places?

Reply:
Yes I agree,  see my latest change.